components/sidebar.jsx

```diff
@@ -124,15 +124,12 @@ export default class Sidebar extends React.Component {
     }
 
     componentDidUpdate(prevProps, prevState) {
-        const allChannels = this.getDisplayedChannels();
-        let isDisplayingChannel = false;
-        for (let i = 0; i < allChannels.length; i++) {
-            if (allChannels[i].id === this.state.activeId) {
-                isDisplayingChannel = true;
-                break;
-            }
-        }
-        if (!isDisplayingChannel) {
+        // if the active channel disappeared (which can happen when dm channels autoclose), go to town square
+        if (this.state.currentTeam === prevState.currentTeam
```

Comment:
Does this line mean that the logic won't be executed when switching teams? Isn't it possible that the target channel has disappeared since the last time that the user looked at `this.state.currentTeam`? And if so, shouldn't we execute this check in that case?